lld/test/ELF/gc-sections-tls.s

```diff
@@ -7,6 +7,11 @@
 
 # ERR: error: {{.*}}.o has an STT_TLS symbol but doesn't have an SHF_TLS section
 
+## As a corner case, when /DISCARD/ is present, demoteLocalSymbolsInDiscardedSections
```

Comment:
Maybe worth putting a FIXME or TODO here so that if this changes then we don't think that this is required behaviour.

Looking back in the history this error was put in to prevent a crash when a symbol was incorrectly typed, so I think not emitting the error is OK in this case.

I think it is illegal to refer to TLS variables using non-TLS relocations anyway. Perhaps in some future patch we can check that.

Reply:
Added a TODO.

Yes, this is pretty much an edge case. We probably only want to add an error if the change does not have performance regression.